### PR TITLE
Enable database rotation for Jakarta Data DDLGen testing

### PR DESCRIPTION
**Pull request by KyleAure**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

- Enable DatabaseRotation for DDL Gen testing
- Create a common DDL Generation helper class
- Execute DDL file on servlet
- Create a database user and schema on databases for testing

### CODE REVIEW

**Comment by KyleAure** [comment]
#build - personal build
#libby - check headers

**Comment by KyleAure** [comment]
DB Rotation Pipeline: https://libh-proxy1.fyre.ibm.com/cognitive-dev/pipelineAnalysis.html?pipelineId=80a50c93-ba04-4f7d-9b41-32b2e27cfd6f

Target locations of links might be accessible only to IBM employees.

**Comment by KyleAure** [comment]
RTC Build: Passed
Pipeline: Failed with 1 known z/OS defect
DB Rotation: Failed with 5 known JPA defects

File: dev/fattest.simplicity/src/componenttest/topology/utils/DDLGenScript.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        }
+
+        /**
+         * Asserts that all the expected DDL files exists in the DDL directory
+         *
+         * @param  expectedFileNames the list of expected file names with .ddl extensions
+         * @return                   this
+         */
+        public DDLGenScriptResult assertDDLFiles(List<String> expectedFileNames) {
+            assertEquals("Expected number of ddl files did not match actual number of files", expectedFileNames.size(), getFileNames());
```
Comment (njr-11) [inline dev/fattest.simplicity/src/componenttest/topology/utils/DDLGenScript.java]: This was comparing a number of files to a list of files. In the suggestion to correct that, I also added the list of observed files to the error message to help with debugging.
```suggestion
            List<String> observedFileNames = getFileNames();
            assertEquals("Expected number of ddl files did not match actual number of files " + observedFileNames,
                         expectedFileNames.size(),
                         observedFileNames.size());
```

**Review by njr-11 — CHANGES_REQUESTED** [review]
This looks great!  I spotted one minor bug in a test utility method and added a suggestion for it.

**Comment by KyleAure** [comment]
#build

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
Pipeline: Failed with 1 known defect
RTC Build: Failed with 1 known build time error

**Pull request merged**